michelson.k

```diff
@@ -865,14 +870,17 @@ module MICHELSON
   syntax String ::= #Blake2BKeyHash(String) [function] // TODO: Blake2B crypto hook.
   rule #Blake2BKeyHash(S) => S
 
+  rule <k> BLAKE2B A => #HandleAnnotations(A) ... </k>
+       <stack> B:MBytesLiteral => #StringToMBytes("0x" +String Blake2B_32(Bytes2String(Int2Bytes(lengthString(#MBytesContent(B)) >>Int 1,#MBytesToInt(B),BE)))) ... </stack>
+
   rule <k> HASH_KEY A => #HandleAnnotations(A) ... </k>
        <stack> #Key(S) => #KeyHash(#Blake2BKeyHash(S)) ... </stack> 
   
   rule <k> SHA256 A => #HandleAnnotations(A) ... </k>
-       <stack> B:MBytes => #SHA256(B) ... </stack>
```

Comment:
These rules probably shouldn't go away permanently, rather the new rules should get the side condition that the B is an MBytesLiteral rather than any other sort of MBytes, and the old rules should get the [owise] attribute.